src/java/main/org/apache/zookeeper/ZooDefs.java

```diff
@@ -74,12 +74,16 @@
 
         public final int createTTL = 21;
 
+        public final int addPersistentWatch = 22;
+
         public final int auth = 100;
 
         public final int setWatches = 101;
 
         public final int sasl = 102;
 
+        public final int setWatches2 = 103;
```

Comment:
@Randgalt im guessing you meant to put "The problem is that it makes 3.5.4 clients incompatible with 3.5.3 servers. That might make sense for 3.6.0 but it's unreasonable for .x.N release." here. I agree totally. This code should stay the same in the 3.5 line but there is no reason it needs to be identical to the 3.6 line which is why i think it should be changed "here".